### PR TITLE
Products pricing section code cleanup

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR addresses a few code issues in the products pricing section:

* ~Removes a bit of unused code~
* Switches from `Currency().formatCurrency`, which is deprecated, to `Currency().formatAmount`
* Renames `priceValidation` to `sanitizeAndSetPrice`, which is more accurate (it was not validating the price, but rather sanitizing it and then setting it) and swaps the order of the `name` and `value`, which is more natural (name, value vs value, name)

This is a follow up to #34382

### How to test the changes in this Pull Request:

1. Go to `Products`> `Add New (MVP` and create a product.
2. Verify that settings the regular and sale price works properly still, including sanitizing of any input value (no regression; functionality should be identical).

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by louwie17 — DISMISSED** [review]
LGTM 🚀 nice work @mattsherman, thanks for keeping it small and not including it as part of other work ( I have something to learn from you :) )
Also appreciate not seeing the giant deprecated warning in the console anymore 👍 👍

File: plugins/woocommerce-admin/client/products/sections/pricing-section.tsx [hunk 1]
```diff
@@ -63,11 +63,6 @@ export const PricingSection: React.FC = () => {
 		return cleanValue;
 	};
 
-	const taxSettingsText =
-		'Per your {{link}}store settings{{/link}}, tax is {{strong}}%sincluded{{/strong}} in the price.';
-	const addNot = pricesIncludeTax ? '' : 'not ';
-	taxSettingsText.replace( taxSettingsText, addNot );
-
```
Comment (octaedro) [inline plugins/woocommerce-admin/client/products/sections/pricing-section.tsx]: We will need the `addNot` to add it when the tax is not included in the price (37-gh-woocommerce/mothra-private). 
The code should look like this:
```
	const addNot = pricesIncludeTax ? '' : 'not ';
	const taxSettingsElement = interpolateComponents( {
		mixedString: sprintf(
			__(
				'Per your {{link}}store settings{{/link}}, tax is {{strong}}%sincluded{{/strong}} in the price.',
				'woocommerce'
			),
			addNot
		),
		components: {
			link: (
				<Link
					href={ `${ ADMIN_URL }admin.php?page=wc-settings&tab=tax` }
					target="_blank"
					type="external"
					onClick={ () => {
						recordEvent( 'product_pricing_list_price_help' );
					} }
				>
					<></>
				</Link>
			),
			strong: <strong />,
		},
	} );
```

I missed that in my original PR.

**Review by octaedro — COMMENTED** [review]
Good job @mattsherman! The code looks good and other than the comment I left this is testing well on my end.

**Review by louwie17 — APPROVED** [review]
Thanks for reverting the one change that @octaedro mentioned, this still tests well.
🚢

**Pull request merged**